lib/builder.js

```diff
@@ -151,7 +152,14 @@ module.exports = class Builder extends EventEmitter {
   // Destructor-like method. Cleanup is synchronous at the moment, but in the
   // future we might change it to return a promise.
   cleanup() {
-    this.builderTmpDirCleanup();
+    const promise = new Promise(resolve => {
+      resolve(this._cancelationRequest);
+    });
+    this._isCleaning = true;
+
+    return promiseFinally(promise, () => this.builderTmpDirCleanup()).finally(() => {
```

Comment:
`finally` doesn't exist in old nodes, we must use `promiseFinally` helper

Reply:
I thought `promiseFinally` returned an object you can call `finally` on...?

Reply:
nope, it doesn't mutate anything, just provides behavior.

Reply:
Fixed